scitt/create_detatched_signed_statement.py

```diff
@@ -0,0 +1,217 @@
+                    KpKty: KtyEC2,
+                    EC2KpCurve: P256,
+                    EC2KpX: x_part,
+                    EC2KpY: y_part,
+                },
+            },
+        },
+        HEADER_LABEL_PAYLOAD_HASH_ALGORITHM: -16,  # for sha256
+        HEADER_LABEL_LOCATION_HINT: location_hint,
+    }
+
+    # now create a sha256 hash of the payload
+    #
+    # NOTE: any hashing algorithm can be used.
+    payload_hash = sha256(payload.encode("utf-8")).digest()
+
+    # create the statement as a sign1 message using the protected header and payload
+    statement = Sign1Message(phdr=protected_header, payload=payload_hash)
+
+    # create the cose_key to sign the statement using the signing key
+    cose_key = {
+        KpKty: KtyEC2,
+        EC2KpCurve: P256,
+        KpKeyOps: [SignOp, VerifyOp],
+        EC2KpD: signing_key.to_string(),
+        EC2KpX: x_part,
+        EC2KpY: y_part,
+    }
+
+    cose_key = CoseKey.from_dict(cose_key)
+    statement.key = cose_key
+
+    # sign and cbor encode the statement.
+    # NOTE: the encode() function performs the signing automatically
+    signed_statement = statement.encode([None])
+
+    return signed_statement
+
+
+def main():
+    """Creates a signed statement"""
+
+    parser = argparse.ArgumentParser(description="Create a signed statement.")
+
+    # signing key file
+    parser.add_argument(
+        "--signing-key-file",
+        type=str,
+        help="filepath to the stored ecdsa P-256 signing key, in pem format.",
+        default="scitt-signing-key.pem",
+    )
+
+    # payload-file (a reference to the file that will become the payload of the SCITT Statement)
+    parser.add_argument(
+        "--payload-file",
+        type=str,
+        help="filepath to the content that will become the payload of the SCITT Statement "
+        "(currently limited to json format).",
+        default="scitt-payload.json",
+    )
+
+    # content-type
+    parser.add_argument(
+        "--content-type",
+        type=str,
+        help="The iana.org media type for the payload",
+        default="application/json",
+    )
+
+    # feed
+    parser.add_argument(
+        "--feed",
+        type=str,
+        help="feed to correlate statements made about an artifact.",
+    )
+
+    # issuer
+    parser.add_argument(
+        "--issuer",
+        type=str,
+        help="issuer who owns the signing key.",
+    )
+
+    # location hint
+    parser.add_argument(
+        "--location-hint",
+        type=str,
+        help="location hint for the original statement that was hashed.",
+    )
+
+    # output file
+    parser.add_argument(
+        "--output-file",
+        type=str,
+        help="name of the output file to store the signed statement.",
+        default="signed-statement.cbor",
+    )
+
+    args = parser.parse_args()
+
+    signing_key = open_signing_key(args.signing_key_file)
+    payload = open_payload(args.payload_file)
+
+    signed_statement = create_signed_statement(
```

Comment:
Can we add the hash algorithm?

Reply:
We can but in the first pass I don't think its best.

This is still a sample script, sha256 is a very common hashing algorithm.

We can stick a large comment saying any hashing algorithm will work. Then its upto the customer to decide which hashing algorithm they use and set

Reply:
resolving as we say in a comment that any hash alg can be used, those using this script as a basis should read this and use the hash alg they want.